aips/aip-103.md

```diff
@@ -0,0 +1,110 @@
+---
+aip: 103
+title: Enable interoperability for Federated Keyless Accounts for the same issuer (user-pool/tenant)
+author: Oliver He (oliver.he@aptoslabs.com)
+Status: Draft # | Last Call | Accepted | Final | Rejected>
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: <Standard (Core, Networking, Interface, Application, Framework) | Informational | Process>
+created: 11/08/2024
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-103.md
+---
+
+# AIP-103 - Enable interoperability for Federated Keyless Accounts for the same issuer (user-pool/tenant)
+
+## Summary
+
+This AIP proposes enabling Aptos Federated Keyless to be interoperable with dApps from the same issuer (user-pool/tenant).
+
+For IAM providers like Auth0 and Cognito, JWT tokens are scoped to a user-pool/tenant via the `iss` field, and they are also scoped to a specific application via the `aud` field.  This means that JWTs from the same issuer but with different `aud` values are from different applications and cannot be used to derive the same Aptos Federated Keyless Account even though they represent the same user identity within the same user-pool/tenant.
+
+Many customers of Auth0 and Cognito have applications with different branding within the same user-pool/tenant ecosystem.  Thus it is natural for such customers to use different application identifiers for their applications for organizational purposes.  This AIP will enable Aptos Federated Keyless Accounts to be interoperable across such applications.
+
+## Impact and risks
+
+- The relaxation of the `aud` field will allow for broader interoperability across applications within the same user-pool/tenant.  This allows for broader adoption of Aptos Federated Keyless Accounts in such user ecosystems.
+
+Risks
+- Developers need to not use `aud`-less accounts when not appropriate. This can be mitigated by the Aptos SDK default behavior so that developers must explicitly enable using `aud`-less accounts.
+- This introduces an additional proving path, one where the `aud` is not checked.  It is important that such proofs are rejected if the account requires `aud` to be present, as encoded in the KeylessPublicKey.
```

Comment:
does it mean `aud_required` should be an onchain config put under the "admin account" of the dapp?
relying on user input is never safe...

Reply:
by "admin account" i mean the jwk addr defined [here](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-96.md)...

Reply:
Well it wouldn't be user input, it would be the dapp developer, unless that's who you mean by user

Reply:
it's eventually up to the user agent (which is under user's control) whether to go aud-less mode, no?
If the dapp developer wants ensure such invalid proofs are rejected, it can only be done on chain?